test/unit/int_algolia/scripts/algolia/helper/productFilter.test.js

```diff
@@ -39,6 +58,130 @@ describe('productFilter.isInStock', () => {
 
         expect(inStock).toBe(false);
     });
+
+    test('Master product with in-stock variant is considered in stock', () => {
+        const masterProduct = new VariantMock();
+        masterProduct.master = true;
```

Comment:
This doesn't feel ok, if there is something that prevents you to use the `MasterProductMock` to represent a master product it's better to fix it.